packages/gatsby-source-graphcms/src/util.js

```diff
@@ -0,0 +1,29 @@
+import R from 'ramda'
+
+
+// If type ends in a non-vowel, we need to append es. Else s.
+export const formatTypeName = t => `all${t}${/s$/.test(t) ? `es` : `s`}`
```

Comment:
Well… what does GraphCMS do? That's what matters. Unless they use pluralize as well you'll just generate a bunch of invalid queries. Can't you get directly from the available types the connection name?